File: py/01A.py
#!/usr/bin/python3


with open('../data/001.txt') as data:
    elfcount = 0
    max_cals = [0, 0]
    temp_cal = 0

    for line in data:
        if line == '\n':
            elfcount += 1
            if temp_cal > max_cals[0]:
                max_cals[0] = temp_cal
                max_cals[1] = elfcount
            temp_cal = 0

        else:
            temp_cal += int(line)

    print(max_cals[0])

File: c/01A.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    FILE *fp;
    fp  = fopen ("../data/001.txt", "r");
    char strbuf[30];
    int calorie_count;
    int elf_cals = 0;
    int max_cals = 0;
    

    while(fgets(strbuf, 30, fp) != NULL){
        if(!strcmp(strbuf, "\n")){
            if(elf_cals > max_cals)
                max_cals = elf_cals;
            elf_cals = 0;
        }else{
            elf_cals += atoi(strbuf);
        }
    }

    printf("%d\n", max_cals);

    fclose(fp);
} 
